addons/website_version/models/version.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+from openerp import fields, models, api
+from openerp.exceptions import Warning
+from openerp.http import request
+from openerp.tools.translate import _
+
+
+class version(models.Model):
+
+    _name = "website_version.version"
+    
+    name = fields.Char(string="Title", required=True)
+    view_ids = fields.One2many('ir.ui.view', 'version_id', string="view_ids", copy=True)
+    website_id = fields.Many2one('website', ondelete='cascade', string="Website")
+    create_date = fields.Datetime('Create Date')
+
+    _sql_constraints = [
+        ('name_uniq', 'unique(name, website_id)', _('You cannot have multiple versions with the same name in the same domain!')),
+    ]
+
+    @api.one
+    def unlink(self):
+        result = self.env['website_version.experiment'].search(['|', ('state', '=', 'running'), ('state', '=', 'paused'), ('experiment_version_ids.version_id', '=', self.id)])
+        if result:
+            raise Warning(_("You cannot delete a version which is in a running or paused experiment."))
+        #To avoid problem when we delete versions in Backend
+        if request:
+            request.session['version_id'] = 0
+        return super(version, self).unlink()
+
+    @api.model
+    def action_publish(self, ids):
+        if self.env.context is None:
+            self.env.context = {}
+        version_id = self.env.context.get('active_id')
+        if version_id:
+            view_ids = self.env['ir.ui.view'].search([('version_id', '=', version_id)])
+            if view_ids:
+                view_ids.publish()
+
+    @api.one
+    def publish_version(self, save_master, copy_master_name):
```

Comment:
Well the code seems complicated just for removing / updating some views ... I won't go into all details but I am quite sure it could be simplified.